README.md

```diff
@@ -6,39 +6,165 @@ Please edit this README.md file with your answers to the worksheet questions.
+
 ## Question 3
 
 ### a
 
+| A	| B	| NOT (A OR B)	| NOT A AND NOT B | 
+|---|---|---|---|
+| TRUE	| TRUE	| FALSE		| FALSE | 
+| TRUE	| FALSE	| FALSE		| FALSE | 
+| FALSE	| TRUE	| FALSE		| FALSE | 
+| FALSE |	FALSE	| TRUE		| TRUE | 
+
 ### b
 
+| A	| B	| NOT (A AND B)	| NOT A OR NOT B | 
+|---|---|---|---|
+| TRUE	| TRUE	| FALSE		| FALSE | 
+| TRUE	| FALSE	| TRUE		| TRUE | 
+| FALSE	| TRUE	| TRUE		| TRUE | 
+| FALSE	| FALSE	| TRUE		| TRUE | 
+
 ### c
 
+| A	| B	| C	| (A AND B) OR (A AND C)	| A AND (B OR C) | 
+|---|---|---|---|---|
+| TRUE	| TRUE	| TRUE	| TRUE			| TRUE | 
+| TRUE	| TRUE	| FALSE	| TRUE			| TRUE |
+| TRUE	| FALSE	| TRUE	| TRUE			| TRUE |
+| TRUE	| FALSE	| FALSE	| FALSE			| FALSE |
+| FALSE	| TRUE	| TRUE	| FALSE			| FALSE |
+| FALSE	| TRUE	| FALSE	| FALSE			| FALSE |
+| FALSE	| FALSE	| TRUE	| FALSE			| FALSE |
+| FALSE	| FALSE	| FALSE	| FALSE			| FALSE |
+
 ### d
 
+| A	| B	| C	| (A OR B) AND (A OR C)	| A OR (B AND C) |
+|---|---|---|---|---|
+| TRUE	| TRUE	| TRUE	| TRUE			| TRUE |
+| TRUE	| TRUE	| FALSE	| TRUE			| TRUE |
+| TRUE	| FALSE	| TRUE	| TRUE			| TRUE |
+| TRUE	| FALSE	| FALSE	| TRUE			| TRUE |
+| FALSE	| TRUE	| TRUE	| TRUE			| TRUE |
+| FALSE	| TRUE	| FALSE	| FALSE			| FALSE |
+| FALSE	| FALSE	| TRUE	| FALSE			| FALSE |
+| FALSE	| FALSE	| FALSE	| FALSE			| FALSE |
+
 ## Question 4
 
 ### a
 
+If we assume that 'file_exists ("a.txt ")' is equal to event A and 'file_exists ("b.txt ")' is equal to event B then both programs can be simplified to identity B in question 3
+
 ### b
 
+If we assume that 'x > 7' is equal to event A, 'type (x) == int' is equal to event B and 'type (x) == float' is equal to event C then both programs can be simplified to identity C in question 3
+
 ### c
 
+If we assume that 'x == 0' is equal to event A and 'y == 0' is equal to event B then we can make a truth table:
+
+| A	| B	| A AND B		| NOT A OR NOT B | 
+|---|---|---|---|
+| TRUE	| TRUE	| TRUE		| FALSE | 
+| TRUE	| FALSE	| FALSE		| TRUE | 
+| FALSE	| TRUE	| FALSE		| TRUE | 
+| FALSE	| FALSE	| FALSE		| TRUE | 
+
+Both if statements always have opposite results (TRUE or FALSE) but since the executed code within the if/else statement is swapped, both programs execute the same code with any inputs as long as they are the same for both programs.
+
 ### d
 
+If we assume that 'x > 10' is equal to event A, 'x > 0' is equal to event B and 'y > 0' is equal to event C then we can make a truth table:
+
+| A	| B	| C	| A OR (B AND C)		| B AND (A OR C) | 
+|---|---|---|---|---|
+| TRUE	| TRUE	| FALSE	| TRUE			| TRUE | 
+| TRUE	| TRUE	| TRUE	| TRUE			| TRUE | 
+| TRUE	| FALSE	| FALSE	| TRUE			| FALSE | 
+| TRUE	| FALSE	| TRUE	| TRUE			| FALSE | 
+| FALSE	| TRUE	| FALSE	| FALSE			| FALSE | 
+| FALSE	| TRUE	| TRUE	| TRUE			| TRUE | 
+| FALSE	| FALSE	| FALSE	| FALSE			| FALSE | 
+| FALSE	| FALSE	| TRUE	| FALSE			| FALSE | 
+
+The two if statements don't always have the same result, meaning that the two programs are actually not equivalent
```

Comment:
The cases where they differ are those where A is TRUE and B is FALSE, so where `x>10` but not `x>0`. However this situation is impossible (a number cannot be both greater than 10 and not greater than 0) so these cases can be ignored, hence the programs are equivalent.